addons/account/models/account_bank_statement_line.py

```diff
@@ -647,26 +647,30 @@ def _retrieve_partner(self):
             sub_queries.append(rf'''
                 {unaccent("%s")} ~* ('^' || (
                    SELECT STRING_AGG(CONCAT('(?=.*\m', chunk[1], '\M)'), '')
-                   FROM regexp_matches({unaccent('name')}, '\w{{3,}}', 'g') AS chunk
+                   FROM regexp_matches({unaccent('partner.name')}, '\w{{3,}}', 'g') AS chunk
                 ))
             ''')
             params.append(text_value)
 
         if sub_queries:
             self.env['res.partner'].flush_model(['company_id', 'name'])
+            self.env['account.move.line'].flush_model(['partner_id', 'company_id'])
             self._cr.execute(
                 '''
-                    SELECT id
-                    FROM res_partner
-                    WHERE (company_id IS NULL OR company_id = %s)
-                        AND name IS NOT NULL
+                    SELECT aml.partner_id
+                    FROM account_move_line aml
+                    JOIN res_partner partner ON
+                        aml.partner_id = partner.id
+                        AND partner.name IS NOT NULL
                         AND (''' + ') OR ('.join(sub_queries) + ''')
```

Comment:
Shouldn't you also check that the partner is active ? (in case the accounting history contains several occurences of the same guy with different objects that got archived over time)